tests/lib/rules/no-const-assign.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview Tests for no-const-assign rule.
+ * @author Toru Nagashima
+ * @copyright 2015 Toru Nagashima. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var eslint = require("../../../lib/eslint");
+var ESLintTester = require("eslint-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var eslintTester = new ESLintTester(eslint);
+eslintTester.addRuleTest("lib/rules/no-const-assign", {
+    valid: [
+        {code: "const x = 0; { let x; x = 1; }", ecmaFeatures: {blockBindings: true}},
+        {code: "const x = 0; function a(x) { x = 1; }", ecmaFeatures: {blockBindings: true}},
+        {code: "const x = 0; foo(x);", ecmaFeatures: {blockBindings: true}},
+        {code: "for (const x in [1,2,3]) { foo(x); }", ecmaFeatures: {blockBindings: true}},
+        {code: "for (const x of [1,2,3]) { foo(x); }", ecmaFeatures: {blockBindings: true, forOf: true}}
+    ],
+    invalid: [
+        {
+            code: "const x = 0; x = 1;",
+            ecmaFeatures: {blockBindings: true},
+            errors: [{message: "`x` is constant.", type: "Identifier"}]
+        },
+        {
+            code: "const x = 0; ({x}) = {x: 1};",
+            ecmaFeatures: {blockBindings: true, destructuring: true},
+            errors: [{message: "`x` is constant.", type: "Identifier"}]
+        },
+        {
+            code: "const x = 0; ({a: x = 1}) = {};",
+            ecmaFeatures: {blockBindings: true, destructuring: true},
+            errors: [{message: "`x` is constant.", type: "Identifier"}]
+        },
+        {
+            code: "const x = 0; x += 1;",
+            ecmaFeatures: {blockBindings: true},
+            errors: [{message: "`x` is constant.", type: "Identifier"}]
+        },
+        {
+            code: "const x = 0; ++x;",
+            ecmaFeatures: {blockBindings: true},
+            errors: [{message: "`x` is constant.", type: "Identifier"}]
+        },
+        {
+            code: "for (const i = 0; i < 10; ++i) { foo(i); }",
```

Comment:
The `ForStatement.update` is evaluated normally, so this `++i` raises a runtime error.